learning-another-language/classes.md

```diff
@@ -0,0 +1,361 @@
+    this.title = title;
+    this.contributor = contributor;
+  }
+}
+
+const chefZine = new Zine('So you want to be a pastry chef', 'Julia Child');
+// > The constructor ran!
+```
+
+**Exercise**:  Try creating an `Animal` class similar to `Zine` above.  The `constructor` should take a `sound` parameter and save it in an instance variable.  Then create an instance of the class.  Verify that it works.  
+
+
+<details>
+  <summary>Click here for a solution</summary>
+
+  ```javascript
+  class Animal {
+    constructor(sound) {
+      this.sound = sound;
+    }
+  }
+
+  const hyena = Animal('heehehehehe hehehe hehehehe!')
+  ```
+</details>
+
+### Instance Methods
+
+Instance methods are created like constructors but with their own names.
+
+```javascript
+const SALESTAX = 0.08;
+
+class Zine {
+  constructor(title, contributor, price) {
+    this.title = title;
+    this.contributor = contributor;
+    this.price = price;
+  }
+
+  totalPrice() {
+    return this.price * (1 + SALESTAX);
+  }
+}
+
+const chefZine = new Zine('So you want to be a pastrychef', 'Julia Child', 1.00);
+console.log(chefZine.totalPrice());
+// > 1.08
+```
+
+**Exercise**:  For our `Animal` class, create a `speak` method which will print the sound to the console.  Try it out to verify that the method is working.  
+
+
+<details>
+  <summary>Click here for a solution</summary>
+
+  ```javascript
+  class Animal {
+    constructor(sound) {
+      this.sound = sound;
+    }
+    speak() {
+      print(sound);
+    }
+  }
+
+  const hyena = Animal('heehehehehe hehehe hehehehe!')
+  heyna.speak();
+  ```
+</details>
+
+### Static Methods
+
+Static methods are equivalent to class methods in Python.  They are attached to the class rather than instances of the class.
+
+```javascript
+class Zine {
+  constructor(title, contributor, price) {
+    this.title = title;
+    this.contributor = contributor;
+    this.price = price;
+  }
+
+  static lowestPrice(zineA, zineB) {
+    if (zineA.price <= zineB.price) {
+      return zineA.title;
+    } else {
+      return zineB.title;
+    }
+  }
+}
+
+const chefZine = new Zine('So you want to be a pastry chef', 'Julia Child', 2.00);
+const linuxZine = new Zine("Linux debugging tools you'll love", 'Evans', 1.00);
+
+console.log(Zine.lowestPrice(chefZine, linuxZine));
+// > Linux debugging tools you'll love
+
+```
+
+Just like a Python class method, the above example calls the static method on the name of the class as `Zine.lowestPrice`.  Trying to call the method with `chefZine.lowestPrice(chefZine, linuxZine);` will result in a TypeError.
+
+**Exercise**:  Add a static method to Animal called `createAnimals`, which takes an array of Strings (sounds) and returns an array of Animals which make those sounds.  
+
+
+<details>
+  <summary>Click here for a solution</summary>
+
+  ```javascript
+  class Animal {
+    constructor(sound) {
+      this.sound = sound;
+    }
+    // speak() removed for brevity
+    static createAnimals(sounds) {
```

Comment:
Note:  We didn't teach static methods in Python...

Reply:
didn't we? in OOP continued? Though, I think for JavaScript `static` is equivalent to `@classmethod` in Python. Do you think we should take it out then, or?

Reply:
It just might bear some connection to what was done in Python then.  They never used `@classmethod` in a project to my recollection.